bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityType.java

```diff
@@ -70,11 +70,11 @@ public class QuantityType<T extends Quantity<T>> extends Number
     public static final QuantityType<Dimensionless> ZERO = new QuantityType<>(0, AbstractUnit.ONE);
     public static final QuantityType<Dimensionless> ONE = new QuantityType<>(1, AbstractUnit.ONE);
 
-    // Regular expression to split unit from value. Split on any blank character, even none (\\s*)
-    // which occurs after a digit (?<=\\d) and before a "unit" character ?=[a-zA-Z°µ\u03BC%']
-    // which itself must not be preceded by plus/minus digit (?![\\+\\-]?\\d).
+    // Regular expression to split unit from value. Split on any blank character, or
+    // between a digit (?<=\\d) and a non-digit character (?=\\D)
+    // which must not be preceded by plus/minus digit (?![\\+\\-]?\\d).
     // The latter would be an exponent from the scalar value.
-    private static final String UNIT_PATTERN = "(?<=\\d)\\s*(?=[a-zA-Z°µ\u03BC%'](?![\\+\\-]?\\d))";
+    private static final String UNIT_PATTERN = "\\s+|(?<=\\d)(?=\\D(?![\\+\\-]?\\d))";
```

Comment:
```suggestion
    private static final Pattern UNIT_PATTERN = Pattern.compile("\\s+|(?<=\\d)(?=\\D(?![\\+\\-]?\\d))");
```

Compiling a regex pattern is very expensive. We should do this once here and then use `UNIT_PATTERN.split(...)` in l. 120 instead of compiling it very time we create a `QuantityType`.

Reply:
Interesting. I couldn't see @wborn's review. I'll add the changes myself.